src/ExcelProvider.Runtime/ExcelProvider.Runtime.fs

```diff
@@ -329,16 +338,27 @@ type Row(documentId, sheetname, rowIndex, getCellValue: int -> int -> obj, colum
             |> Seq.map (fun kvp -> kvp.Key)
             |> Seq.tryFind (fun header -> String.Equals(header, columnName, StringComparison.OrdinalIgnoreCase))
             |> function
-                | Some header -> sprintf "ExcelProvider: Column \"%s\" was not found. Did you mean \"%s\"?" columnName header
+                | Some header ->
+                    sprintf "ExcelProvider: Column \"%s\" was not found. Did you mean \"%s\"?" columnName header
                 | None -> sprintf "ExcelProvider: Column \"%s\" was not found." columnName
             |> failwith
 
     member this.TryGetValue<'a> (columnIndex: int) columnName =
         let value = this.GetValue columnIndex
 
         try
-            value :?> 'a
-        with :? InvalidCastException ->
+            match value with
+            | null -> value :?> 'a
+            | _ when typeof<'a> = typeof<string> -> (box (string value)) :?> 'a
+            //| :? string as valueStr when typeof<'a> = typeof<double> -> (box (Double.Parse valueStr)) :?> 'a
+            | :? double as valueDbl when typeof<'a> = typeof<DateTime> -> (box (DateTime.FromOADate valueDbl)) :?> 'a
+            | :? string as valueStr when typeof<'a> = typeof<DateTime> -> (box (DateTime.Parse valueStr)) :?> 'a
+            | _ -> value :?> 'a
+        with
+        | :? InvalidCastException
+        | :? ArgumentException
+        | :? ArgumentNullException
+        | :? InvalidCastException ->
```

Comment:
It might be a bit cleaner to split out/handle each case a bit differently.

The default error message seems fine, although more information (for example, parsing doubles as OA dates by default) might be more useful to the end user.